### DISCUSSION TITLE
Is it safe to use get_request() in plugin signal?

### DISCUSSION THREAD

**Discussion by Selfmade-RuLeZ** [netbox-community/netbox#10651] [Q&A] [answered]
Hi guys,

I'm currently developing a plugin and I need to make a conjunction between my model and the user who created an object of an existing model (VirtualMachine). 

I get the signal of the created VirtualMachine and can create my new object. I even successful tried to get the user of the request by using `get_request()`. Now I have safety concerns about the usage of `get_request()` as it is a threaded variable. Do I have to implement something else or even drop this approach or just use it like in `signals.py`?

https://github.com/netbox-community/netbox/blob/f94c1e91ea20c805528125407193371476ccbcd8/netbox/extras/signals.py#L33

Thanks a lot!

**Comment by jeremystretch** [comment, marked as answer]
As a rule, never assume any undocumented component of the NetBox code base is safe to use. Non-public parts of the plugins API are subject to change without warning. If there's something in particular you'd like to see officially supported, please consider opening a [feature request](https://github.com/netbox-community/netbox/issues/new?assignees=&labels=type%3A+feature&template=feature_request.yaml).
